File: Data Structures & Algorithms/maximum-subarray/submission-0.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        ans = nums[0]
        current_sum = 0

        for n in nums:
            if current_sum < 0:
                current_sum = n
            else:
                current_sum += n

            ans = max(ans, current_sum)

        
        return ans

File: Data Structures & Algorithms/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/submission-0.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr, int k, int threshold) {
        int L = 0;
        long long current_sum = 0, target = threshold * k;
        int ans = 0;

        for (int R = 0; R < arr.size(); R++) {
            current_sum += arr[R];
            if (R - L + 1 > k) {
                current_sum -= arr[L];
                L++;
            }
            
            if (R - L + 1 == k) {
                if (current_sum >= target) ans++;
            }
        }

        return ans;
    }
};